numpy/_core/arrayprint.py

```diff
@@ -295,25 +279,21 @@ def set_printoptions(precision=None, threshold=None, edgeitems=None,
     array([ 0.  ,  1.11,  2.22, ...,  7.78,  8.89, 10.  ])
 
     """
-    opt = _make_options_dict(precision, threshold, edgeitems, linewidth,
-                             suppress, nanstr, infstr, sign, formatter,
-                             floatmode, legacy, override_repr)
-    # formatter and override_repr are always reset
-    opt['formatter'] = formatter
-    opt['override_repr'] = override_repr
-    _format_options.update(opt)
-
-    # set the C variable for legacy mode
-    if _format_options['legacy'] == 113:
-        set_legacy_print_mode(113)
-        # reset the sign option in legacy mode to avoid confusion
-        _format_options['sign'] = '-'
-    elif _format_options['legacy'] == 121:
-        set_legacy_print_mode(121)
-    elif _format_options['legacy'] == 125:
-        set_legacy_print_mode(125)
-    elif _format_options['legacy'] == sys.maxsize:
-        set_legacy_print_mode(0)
+    new_opt = _make_options_dict(precision, threshold, edgeitems, linewidth,
+                            suppress, nanstr, infstr, sign, formatter,
+                            floatmode, legacy)
+    # formatter is always reset
+    new_opt['formatter'] = formatter
+    new_opt['override_repr'] = override_repr
+
+    updated_opt = format_options.get().copy()
```

Comment:
How about `updated_opt = format_options.get() | new_opt` (this makes a new `dict` too, so avoids the `copy`)